[
    {
        "number": 2,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "scikit-learn"
            },
            "manifest_path": "setup.py",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-jjw5-xxj6-pcv5",
            "cve_id": "CVE-2020-13092",
            "summary": "scikit-learn Deserialization of Untrusted Data",
            "description": "scikit-learn (aka sklearn) through 0.23.0 can unserialize and execute commands from an untrusted file that is passed to the `joblib.load()` function, if `__reduce__` makes an `os.system call`.\nNOTE: third parties dispute this issue because the joblib.load() function is documented as unsafe and it is the user's responsibility to use the function in a secure manner.",
            "severity": "critical",
            "identifiers": [
                {
                    "value": "GHSA-jjw5-xxj6-pcv5",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2020-13092",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2020-13092"
                },
                {
                    "url": "https://github.com/0FuzzingQ/vuln/blob/master/sklearn%20unserialize.md"
                },
                {
                    "url": "https://scikit-learn.org/stable/modules/model_persistence.html#security-maintainability-limitations"
                },
                {
                    "url": "https://github.com/advisories/GHSA-jjw5-xxj6-pcv5"
                }
            ],
            "published_at": "2022-05-24T17:17:59Z",
            "updated_at": "2024-04-22T22:55:45Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "scikit-learn"
                    },
                    "severity": "critical",
                    "vulnerable_version_range": "<= 0.23.0",
                    "first_patched_version": null
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
                "score": 9.8
            },
            "cwes": [
                {
                    "cwe_id": "CWE-502",
                    "name": "Deserialization of Untrusted Data"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "scikit-learn"
            },
            "severity": "critical",
            "vulnerable_version_range": "<= 0.23.0",
            "first_patched_version": null
        },
        "url": "https://api.github.com/repos/username/snips-nlu/dependabot/alerts/2",
        "html_url": "https://github.com/username/snips-nlu/security/dependabot/2",
        "created_at": "2024-05-24T01:33:41Z",
        "updated_at": "2024-05-24T01:33:41Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 1,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "future"
            },
            "manifest_path": "setup.py",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-v3c5-jqr6-7qm8",
            "cve_id": "CVE-2022-40899",
            "summary": "Python Charmers Future denial of service vulnerability",
            "description": "An issue discovered in Python Charmers Future 0.18.2 and earlier allows remote attackers to cause a denial of service via crafted Set-Cookie header from malicious web server. This issue has been patched in version 0.18.3.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-v3c5-jqr6-7qm8",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2022-40899",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-40899"
                },
                {
                    "url": "https://github.com/python/cpython/pull/17157"
                },
                {
                    "url": "https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215"
                },
                {
                    "url": "https://pypi.org/project/future/"
                },
                {
                    "url": "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"
                },
                {
                    "url": "https://github.com/PythonCharmers/python-future/commit/c91d70b34ef0402aef3e9d04364ba98509dca76f"
                },
                {
                    "url": "https://github.com/PythonCharmers/python-future/pull/610"
                },
                {
                    "url": "https://github.com/advisories/GHSA-v3c5-jqr6-7qm8"
                }
            ],
            "published_at": "2022-12-23T00:30:23Z",
            "updated_at": "2023-01-28T05:06:22Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "future"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "<= 0.18.2",
                    "first_patched_version": {
                        "identifier": "0.18.3"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": []
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "future"
            },
            "severity": "high",
            "vulnerable_version_range": "<= 0.18.2",
            "first_patched_version": {
                "identifier": "0.18.3"
            }
        },
        "url": "https://api.github.com/repos/username/snips-nlu/dependabot/alerts/1",
        "html_url": "https://github.com/username/snips-nlu/security/dependabot/1",
        "created_at": "2024-05-24T01:33:41Z",
        "updated_at": "2024-05-24T01:33:41Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    }
]